x/farm/abci.go

```diff
@@ -0,0 +1,23 @@
+package farm
+
+import (
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmosquad-labs/squad/v3/x/farm/keeper"
+	"github.com/cosmosquad-labs/squad/v3/x/farm/types"
+)
+
+func BeginBlocker(ctx sdk.Context, k keeper.Keeper) {
+	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyBeginBlocker)
+
+	if err := k.TerminateEndedPlans(ctx); err != nil {
+		panic(err)
+	}
+	if err := k.AllocateRewards(ctx); err != nil {
+		panic(err)
+	}
+	k.SetLastBlockTime(ctx, ctx.BlockTime())
```

Comment:
mint module also records `LastBlockTime` in every begin block, how about using `LastBlocktime` from mint module instead of writing the same information twice?

Reply:
That's a possible option, but it creates a strong coupling between two modules which is a thing we should avoid.
For example, if we use only the farm module not the mint module in the future, this module will not work.